AirSkyBoat/AirSkyBoat
# Fixes cleanup script for dyna

fixes cleanup script for dyna




**_I affirm:_**


## Please enter a player-facing description


Players should no longer have a dynamis hourglass expire immediately after trading.
A bug that was allowing the 60s timer for zone cleanup to be bypassed has been fixed.

*Note: As far as players are concerned, they only know that its a 60s timer and not 90s*

## What does this pull request do? (Please be technical)


Currently, the zone cleanup script fires multiple times when a zone expires.
When the timer expires, the cleanup script is fired off every 30s because tracking for when to fire the cleanup uses local vars, which are then reset when the cleanup script fires, thus allowing it to fire the cleanup script again.

This should fix the repeated calls to the zone cleanup function by doing the following:
Sets a server variable for tracking cleanup per zone to ensure the script only fires once
Fixes logical errors in the era_dynamis.lua script that was causing the cleanup script to fire off multiple times.
Removed an unnecessary override that was causing the 90s Cleanup Timer to be bypassed.
Fixed the "time remaining" message from showing -4574398534 minute when the zone expires.

## Steps to test these changes


- Edit era_dynamis.lua and place a print in the cleanup_dynamis function (so you can see how many times it fires off when a zone expires)
- `!additem timeless_hourglass`
- Trade the hourglass to start a dynamis zone

Time Expiration
 - `!adddynatime [zone] -60
 - Allow the zone to expire and watch the map process log
 - Note that the cleanup script is only fired once

No Player in Zone
 - Toss your hourglass after entering and wait for your registration to expire
 - Wait 15 minutes for the zone to expire due to no players in zone
 - Note that the cleanup script is only fired once


## Special Deployment Considerations




Requires the server to be reset to take effect completely due to the removal of a module override function.
